Storage/container/one_to_many.h

```diff
@@ -63,47 +64,82 @@ class GenericOneToMany {
   // Adds specified object and overwrites existing one if it has the same row and col.
   // Removes all other existing objects with the same col.
   void Add(const T& object) {
+    const auto now = current::time::Now();
     const auto row = sfinae::GetRow(object);
     const auto col = sfinae::GetCol(object);
     const auto key = std::make_pair(row, col);
-    const auto cit = map_.find(key);
-    if (cit != map_.end()) {
-      const T& previous_object = *(cit->second);
-      journal_.LogMutation(UPDATE_EVENT(object),
-                           [this, key, previous_object]() { DoAdd(key, previous_object); });
+    const auto map_cit = map_.find(key);
+    const auto lm_cit = last_modified_.find(key);
+    if (map_cit != map_.end()) {
+      const T& previous_object = *(map_cit->second);
+      assert(lm_cit != last_modified_.end());
+      const auto previous_timestamp = lm_cit->second;
+      journal_.LogMutation(UPDATE_EVENT(now, object),
+                           [this, key, previous_object, previous_timestamp]() {
+                             DoAddWithLastModified(previous_timestamp, key, previous_object);
```

Comment:
Here and above, `DoUpdateWithLastModified` is probably a better name -- it's in fact reverting the record, not adding it.